### ISSUE TITLE
Breadcrumb to display header of a long code block

### ISSUE THREAD

**Issue by rakanalh** [emacs-lsp/lsp-mode#4488]
Hello,

After checking out the documentation, it seems like the breadcrumb header is only able to display the current module / method names.

However, i would like to be able to configure it to display the header of a long block as well.

An example of this would be similar to Neovim / Mason / Rustacean:
![image](https://github.com/emacs-lsp/lsp-mode/assets/195829/db231976-1459-4f83-8a30-8a8e41088fc6)

So basically what happens is the following:
Breadcrumb would show:
- Nothing if the cursor is within a method that is completely visible on the screen
- `impl Module -> method` if the method is partially visible on the screen
- `loop / while / for / if` if the cursor is within a method / func that contains these huge blocks of code and is partially visible on the screen.
